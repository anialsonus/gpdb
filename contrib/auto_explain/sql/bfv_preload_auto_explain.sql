
-- start_ignore
\! gpconfig -c shared_preload_libraries -v 'auto_explain';
\! gpstop -raiq;
\c
-- end_ignore

SET auto_explain.log_min_duration = 0;
SET CLIENT_MIN_MESSAGES = LOG;

-- auto_axplain must work only at coordinator with Gp_role is GP_ROLE_DISPATCH
-- check that auto_explain doesn't work on segments which are executors.
-- Query 'CREATE TABLE t1 as select generate_series(1, 1)' and query 'CREATE TABLE t1 as select generate_series(1, 10*1000*1000)'
-- are equally got segfault because it has intermediate executor's slice:
--   Result
--     Output: (generate_series(1, 1))
--       ->  Redistribute Motion 3:3  (slice1; segments: 3)

CREATE TABLE t1 as select generate_series(1, 1);
DROP TABLE t1;

-- check that auto_explain doesn't work on coordinator with Gp_role is not a GP_ROLE_DISPATCH
-- Query 'SELECT count(1) from (select i from t1 limit 10) t join t2 using (i)' generate executor's slice on coordinator:
--             ->  Redistribute Motion 1:3  (slice2)
--                   Output: t1.i
--                   Hash Key: t1.i
--                   ->  Limit
--                         Output: t1.i
--                         ->  Gather Motion 3:1  (slice1; segments: 3)
-- IMPORTANT: ./configure with --enable-orca

CREATE TABLE t1(i int);
CREATE TABLE t2(i int);
SELECT count(1) from (select i from t1 limit 10) t join t2 using (i);
DROP TABLE t1;
DROP TABLE t2;

-- start_ignore
\! gpconfig -r shared_preload_libraries;
\! gpstop -raiq;
-- end_ignore
