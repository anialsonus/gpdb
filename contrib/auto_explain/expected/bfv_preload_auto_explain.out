-- start_ignore
\! gpconfig -c shared_preload_libraries -v 'auto_explain';
20220719:16:51:22:051096 gpconfig:evgeniy-pc:evgeniy-[INFO]:-completed successfully with parameters '-c shared_preload_libraries -v auto_explain'
\! gpstop -raiq;
\c
-- end_ignore
SET auto_explain.log_min_duration = 0;
SET CLIENT_MIN_MESSAGES = LOG;
-- auto_axplain must work only at coordinator with Gp_role is GP_ROLE_DISPATCH
-- check that auto_explain doesn't work on segments which are executors.
-- Query 'CREATE TABLE t1 as select generate_series(1, 1)' and query 'CREATE TABLE t1 as select generate_series(1, 10*1000*1000)'
-- are equally got segfault because it has intermediate executor's slice:
--   Result
--     Output: (generate_series(1, 1))
--       ->  Redistribute Motion 3:3  (slice1; segments: 3)
CREATE TABLE t1 as select generate_series(1, 1);
LOG:  statement: CREATE TABLE t1 as select generate_series(1, 1);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'generate_series' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
LOG:  duration: 0.002 ms  plan:
Query Text: CREATE TABLE t1 as select generate_series(1, 1);
Redistribute Motion 1:3  (slice1; segments: 1)  (cost=0.00..5.01 rows=1000 width=0)
  Hash Key: (generate_series(1, 1))
  ->  Result  (cost=0.00..5.01 rows=1000 width=0)
DROP TABLE t1;
LOG:  statement: DROP TABLE t1;
-- check that auto_explain doesn't work on coordinator with Gp_role is not a GP_ROLE_DISPATCH
-- Query 'SELECT count(1) from (select i from t1 limit 10) t join t2 using (i)' generate executor's slice on coordinator:
--             ->  Redistribute Motion 1:3  (slice2)
--                   Output: t1.i
--                   Hash Key: t1.i
--                   ->  Limit
--                         Output: t1.i
--                         ->  Gather Motion 3:1  (slice1; segments: 3)
-- IMPORTANT: ./configure with --enable-orca
CREATE TABLE t1(i int);
LOG:  statement: CREATE TABLE t1(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE t2(i int);
LOG:  statement: CREATE TABLE t2(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
SELECT count(1) from (select i from t1 limit 10) t join t2 using (i);
LOG:  statement: SELECT count(1) from (select i from t1 limit 10) t join t2 using (i);
LOG:  duration: 2.986 ms  plan:
Query Text: SELECT count(1) from (select i from t1 limit 10) t join t2 using (i);
Aggregate  (cost=1436.96..1436.97 rows=1 width=8)
  ->  Gather Motion 3:1  (slice3; segments: 3)  (cost=1436.90..1436.95 rows=1 width=8)
        ->  Aggregate  (cost=1436.90..1436.91 rows=1 width=8)
              ->  Hash Join  (cost=0.74..1434.49 rows=321 width=0)
                    Hash Cond: (t2.i = t1.i)
                    ->  Seq Scan on t2  (cost=0.00..1063.00 rows=32100 width=4)
                    ->  Hash  (cost=0.61..0.61 rows=4 width=4)
                          ->  Redistribute Motion 1:3  (slice2; segments: 1)  (cost=0.00..0.61 rows=10 width=4)
                                Hash Key: t1.i
                                ->  Limit  (cost=0.00..0.31 rows=10 width=4)
                                      ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..0.31 rows=10 width=4)
                                            ->  Limit  (cost=0.00..0.11 rows=4 width=4)
                                                  ->  Seq Scan on t1  (cost=0.00..1063.00 rows=32100 width=4)
 count 
-------
     0
(1 row)

DROP TABLE t1;
LOG:  statement: DROP TABLE t1;
DROP TABLE t2;
LOG:  statement: DROP TABLE t2;
-- start_ignore
\! gpconfig -r shared_preload_libraries;
20220719:16:51:27:051682 gpconfig:evgeniy-pc:evgeniy-[INFO]:-completed successfully with parameters '-r shared_preload_libraries'
\! gpstop -raiq;
-- end_ignore
